presto-tests/src/test/java/com/facebook/presto/tests/TestGracefulShutdown.java

```diff
@@ -99,13 +116,13 @@ public void testShutdown()
                 assertEquals(info.getState(), FINISHED);
             }
 
-            TestShutdownAction shutdownAction = (TestShutdownAction) worker.getShutdownAction();
+            TestShutdownAction shutdownAction = (TestShutdownAction) testServer.getShutdownAction();
             shutdownAction.waitForShutdownComplete(SHUTDOWN_TIMEOUT_MILLIS);
-            assertTrue(shutdownAction.isWorkerShutdown());
+            assertTrue(shutdownAction.isShutdown());
         }
     }
 
-    @Test(expectedExceptions = UnsupportedOperationException.class)
+    @Test
```

Comment:
Please add a max time limit to this test

Reply:
Makes sense, added timeout for the test.